picard/ui/metadatabox/__init__.py

```diff
@@ -274,8 +274,14 @@ def _copy_value(self):
                 value = self.tag_diff.orig[tag]
             elif column == self.COLUMN_NEW:
                 value = self.tag_diff.new[tag]
+
             if tag == '~length':
-                value = [format_time(value or 0), ]
+                try:
+                    value = float(value)
+                    value = [format_time(value or 0), ]
+                except (TypeError, ValueError):
+                    value = ['']
```

Comment:
This is an interesting issue, and I think it runs even deeper. `value` nearly always is a list of strings. If you copy other values than length it is a list (no matter if the tag contains one or multiple entries). If you use copy on multiple selected files for a tag which has different values in the selected items, than `value` is also a list with a single empty string (`[""]`).

Only when copying the length it isn't a list but a single string. This already is odd, not sure why it isn't consistently a list of strings. @zas any idea? Should we change this?

Also yes, we could extend this to also handle copying of multiple distinct values on multiple selected items. But that's entirely separate from the bugfix :)

Reply:
> Only when copying the length it isn't a list but a single string. This already is odd, not sure why it isn't consistently a list of strings. @zas any idea? Should we change this?

Yes, this is something weird in the code and we have many places where it leads to special handling. It isn't very logical and I'm not sure from where it comes.
Changing it will need a lot of care, but IMHO this is something we should do.



> Also yes, we could extend this to also handle copying of multiple distinct values on multiple selected items.

Yes, separate PR. But the question above is related, one tag value is a list of strings, so multiple objects tag value is a list of lists of strings (and no exception for `~length`).